pkg/webhook/validator.go

```diff
@@ -567,23 +576,25 @@ func ValidatePolicyAttestationsForAuthority(ctx context.Context, ref name.Refere
 			va, err := validAttestationsWithFulcio(ctx, ref, fulcioroot, rekorClient, authority.Keyless.Identities, remoteOpts...)
 			if err != nil {
 				logging.FromContext(ctx).Errorf("failed validAttestationsWithFulcio for authority %s with fulcio for %s: %v", name, ref.Name(), err)
-				return nil, fmt.Errorf("validate signatures with fulcio: %w", err)
+				return nil, fmt.Errorf("keyless validation failed for authority %s for %s: %w", name, ref.Name(), err)
 			}
 			verifiedAttestations = append(verifiedAttestations, va...)
 		}
 	}
+
 	// If we didn't get any verified attestations either from the Key or Keyless
 	// path, then error out
 	if len(verifiedAttestations) == 0 {
-		logging.FromContext(ctx).Errorf("no valid attestations found with authority %s for %s", name, ref.Name())
-		return nil, fmt.Errorf("no valid attestations found with authority %s for %s", name, ref.Name())
+		logging.FromContext(ctx).Errorf("no valid attestations found for authority %s for %s", name, ref.Name())
+		return nil, fmt.Errorf("%w for authority %s for %s", cosign.ErrNoMatchingAttestations, name, ref.Name())
```

Comment:
Pretty sure this too is unreachable, but since we have the loop and all I opted for wrapping the same error here instead.